core/genesis.go

```diff
@@ -141,6 +141,15 @@ func (e *GenesisMismatchError) Error() string {
 	return fmt.Sprintf("database contains incompatible genesis (have %x, new %x)", e.Stored, e.New)
 }
 
+func (e *GenesisMismatchError) Is(target error) bool {
+	gme, ok := target.(*GenesisMismatchError)
+	if !ok {
+		return false
+	}
+	return bytes.Equal(e.Stored[:], gme.Stored[:]) &&
+		bytes.Equal(e.New[:], gme.New[:])
+}
+
```

Comment:
Is this whole thing needed? I'm thinking we can replace it with 
```
var ErrGenesisMismatch = errors.New("genesis mismatch")
```
And in the places where we now instantiate it, we simply do 
```
return fmt.Errorf("%w: have %x, new %x", ErrGenesisMismatch)
```
And then we can just do the generic `errors.Is` for it, instead of implementing custom `Is` support